src/cpl/nuopc/rof_import_export.F90

```diff
@@ -488,169 +522,100 @@ end subroutine SetScalarField
   end subroutine fldlist_realize
 
   !===============================================================================
-
-  subroutine state_getimport(state, fldname, begr, endr, output, do_sum, rc)
+  subroutine state_getimport(state, fldname, begr, endr, output, areacor, rc)
 
     ! ----------------------------------------------
     ! Map import state field to output array
     ! ----------------------------------------------
 
+    use ESMF, only : ESMF_StateGet, ESMF_FieldGet, ESMF_Field
+
     ! input/output variables
     type(ESMF_State)    , intent(in)    :: state
     character(len=*)    , intent(in)    :: fldname
     integer             , intent(in)    :: begr
     integer             , intent(in)    :: endr
     real(r8)            , intent(out)   :: output(begr:endr)
-    logical, optional   , intent(in)    :: do_sum
+    real(r8), optional  , intent(in)    :: areacor(:)
```

Comment:
I just noticed this possible issue, both in this PR (here and in state_setexport) and in the corresponding MOSART PR (https://github.com/ESCOMP/mosart/pull/40): Here we have an optional argument that appears before a required argument in the argument list. Do all compilers allow that? Even if they do, it feels confusing and error-prone to do that.

My first inclination was to make this a required argument, since it seems like it is always passed in practice. But actually, I wonder if a better solution is to not have this be an argument at all, but instead have state_getimport directly use the module variable `med2mod_areacor` and state_setexport directly use `mod2med_areacor`. That feels less error-prone in the future, because you wouldn't need to worry about actually using the wrong one in a given call.

Reply:
Yes, good point. Really areacor should be a required argument -- we would never want to run without it right? It's hardcoded into the MCT cap. I can't think of a reason you'd want to remove it, unless it's just so you can show answers are identical to the previous version if you turn it off.